apps/src/code-studio/initializeBlockPreview.js

```diff
@@ -0,0 +1,36 @@
+function update(blockSpace, container, editor) {
+  try {
+    // populate with new xml; because we do this every time the editor
+    // updates, there is the potential for many different kinds of
+    // errors here as we will end up updating as the levelbuilder is
+    // typing out the new content. Because of that and the fact that
+    // this is simply an informative but not functional view, we simply
+    // catch and ignore all errors.
+    var xml = $.parseXML(editor.getValue()).childNodes[0];
+    blockSpace.clear();
+    Blockly.Xml.domToBlockSpace(blockSpace, xml);
+  } catch (e) {
+    return;
+  }
+
+  // resize
+  var metrics = blockSpace.getMetrics();
+  var height = metrics.contentHeight + metrics.contentTop;
+  container.style.height = height + "px";
+}
+
+module.exports = function (editor, container) {
+  var xml = $.parseXML(editor.getValue() || "<xml></xml>").childNodes[0];
+  var blockSpace = Blockly.BlockSpace.createReadOnlyBlockSpace(container, xml, {
```

Comment:
were your developer tools open and causing the available horizontal space for blockly to be zero? If so, that happens on regular Code Studio blockly as well.